injected/src/content-feature.js

```diff
@@ -146,6 +146,26 @@ export default class ContentFeature {
 
     /**
      * Return a specific setting from the feature settings
+     * If the "settings" key within the config has a "domains" key, it will be used to override the settings.
+     * This uses JSONPatch to apply the patches to settings before getting the setting value.
+     * For example.conm getFeatureSettings('val') will return 1:
+     * ```json
+     *  {
+     *      "settings": {
+     *         "domains": [
+     *             {
+     *                "domain": "example.com",
+     *                "patchSettings": [
+     *                    { "op": "replace", "path": "/val", "value": 1 }
+     *                ]
+     *             }
+     *         ]
+     *      }
+     *  }
+     * ```
+     * "domain" can either be a string or an array of strings.
+
+     * For boolean states you should consider using getFeatureSettingEnabled.
```

Comment:
I tried importing the type from the config to apply to: `this.matchDomainFeatureSetting('domains')` but TS isn't having it.